src/helperFunctions/fetchIdleUsers.ts

```diff
@@ -0,0 +1,19 @@
+import { IDLE_USERS_URL } from '@/components/constants/url';
+import fetch from '@/helperFunctions/fetch';
+import { IdleUser } from '@/interfaces/idleUser.type';
+
+const fetchIdleUsers = async (): Promise<[string[], boolean]> => {
```

Comment:
The `boolean` value is an error flag, it is `false` where there's no error, `true` otherwise.  It can be used inside the page to show errors if any (`setError(boolean)`)

Reply:
do we need this as the code inside the page in anyway wrapped in try-catch block so if something blows up here the code inside catch block will be triggered where we can have `setError(true)`

Reply:
It is not necessary to have it as the code inside the page is anyway wrapped in try-catch block. But this way, it is easier to handle types, like we might need to do type assertion when using in the page as the return types of  idle users array and error is different. 



```tsx 
// When using inside page
const idleUserNames = (await fetchIdleUsers()) as string[]
```

I would be happy to hear your thoughts and better ways to implement this

Reply:
That would not be required as you already have a return type assigned to `fetchIdleUsers` function. 

Example [link](https://www.typescriptlang.org/play?#code/MYewdgzgLgBMCu0QFsBiBTKwAWMC8MY6A7jAAoBOKAlhOgBT0XoQgA2AbugDQzMBW6YFACU+AHwwA3gCgY8uOGh8AhmAAm+GAFkVUbADoAZmxAgK9XfoMU16lPTEAqGAEYADCLkLqRmEzsYSQBWEVkFCL4Wdi56AG0AchUAI2B1BN4E9CMAc2wEgF0vCIBfGW95ASEoegAiVBVqNnRNKBAYUxVNdT0VWq8SrxlQSFgjTBwASXV0FTYAVToKCC0VCABPMGB-EQAuSho6AB5oCmowHLiCyTxJcIUoCnX7yMVRmB6oFVXiRtgEJBoCa4NYwU7nS4FCqRZhQeAUMAfXrQkrAPQ4ejoMLQiKw+GIq4omRlYZKWCddQAEV6qw2Wx2EmkMGhI2Un2+BBUv2oY2B01mCyWEEcLKU7HQBlMOXoACkAMoAeQAcgZwRdfOt6OzeGB4Gw2LwAEwiAblCnUr6OIA)

Reply:
With the current approach that you shared, we will not be able to set an error message inside the page right? It is just an empty page when there's an error.

Reply:
Sounds good